nca/NetworkConfig/NetworkLayer.py

```diff
@@ -77,21 +84,19 @@ def add_policy(self, policy, layer_name):
         if self.default_layer in self and not self[self.default_layer].policies_list:
             del self[self.default_layer]
 
-    def does_contain_single_layer(self, layer_name):
+    def does_contain_only_gateway_layers(self):
         """
-        Checks if the given layer is the only layer in the map.
-        :param NetworkLayerName layer_name: the layer to check
-        :return: True if the layer is the only layer in the map, False otherwise
+        Checks if the map contains only gateway layers.
+        :return: True if the map contains only gateway layers, False otherwise
         """
-        return len(self) == 1 and list(self.keys())[0] == layer_name
+        return set(self.keys()).issubset({NetworkLayerName.K8sGateway, NetworkLayerName.IstioGateway})
 
-    def does_contain_layer(self, layer_name):
+    def does_contain_istio_layers(self):
         """
-        Checks if the given layer is in the map.
-        :param NetworkLayerName layer_name: the layer to check
-        :return: True if the layer is in the map
+        Checks if any of Istio layers is in the map.
+        :return: True if any of Istio layers is in the map, False otherwise
         """
-        return layer_name in self
+        return {NetworkLayerName.Istio, NetworkLayerName.IstioGateway} & set(self.keys())
```

Comment:
this returns a set intersection and not a bool result

Reply:
Fixed